pkg/common/unittestcommon/utils.go

```diff
@@ -320,3 +327,123 @@ func (c *FakeK8SOrchestrator) GetPVNameFromCSIVolumeID(volumeID string) (string,
 func (c *FakeK8SOrchestrator) InitializeCSINodes(ctx context.Context) error {
 	return nil
 }
+
+// configFromVCSim starts a vcsim instance and returns config for use against the
+// vcsim instance. The vcsim instance is configured with an empty tls.Config.
+func configFromVCSim(vcsimParams VcsimParams, isTopologyEnv bool) (*config.Config, func()) {
+	return configFromVCSimWithTLS(new(tls.Config), vcsimParams, true, isTopologyEnv)
+}
+
+// configFromVCSimWithTLS starts a vcsim instance and returns config for use
+// against the vcsim instance. The vcsim instance is configured with a
+// tls.Config. The returned client config can be configured to allow/decline
+// insecure connections.
+func configFromVCSimWithTLS(tlsConfig *tls.Config, vcsimParams VcsimParams, insecureAllowed bool,
+	isTopologyEnv bool) (*config.Config, func()) {
+	cfg := &config.Config{}
+	model := simulator.VPX()
+	// Use user specified values for fields like datacenters, clusters, hosts, VMs, datastores etc.
+	if vcsimParams.Datacenters != VCSimDefaultDatacenters {
+		model.Datacenter = vcsimParams.Datacenters
+	}
+	if vcsimParams.Clusters != VCSimDefaultClusters {
+		model.Cluster = vcsimParams.Clusters
+	}
+	if vcsimParams.HostsPerCluster != VCSimDefaultHostsPerCluster {
+		model.ClusterHost = vcsimParams.HostsPerCluster
+	}
+	if vcsimParams.StandaloneHosts != VCSimDefaultStandalonHosts {
+		model.Host = vcsimParams.StandaloneHosts
+	}
+	if vcsimParams.Datastores != VCSimDefaultDatastores {
+		model.Datastore = vcsimParams.Datastores
+	}
+	if vcsimParams.VMsPerCluster != VCSimDefaultVMsPerCluster {
+		model.Machine = vcsimParams.VMsPerCluster
+	}
+
+	// Currently the simulated vc has a version of 6.5.0, modifying service content to 7.0.3
+	// TODO: update in govmomi
+	serviceContent := vpx.ServiceContent
+	serviceContent.About.Version = "7.0.3"
```

Comment:
Can we supply versions as argument or make it configurable?
We may have requirement to get vCenter 8.0 model?

Reply:
Made changes to make versions configurable. Verified that it works even after giving versions like 8.0.3 etc.